code/game/machinery/traitorous_cloner.dm

```diff
@@ -0,0 +1,360 @@
+		podf = locate(/obj/machinery/clonepod/traitorous, get_step(src, direction))
+		if (!isnull(podf) && podf.is_operational())
+			AttachCloner(podf)
+
+/obj/machinery/computer/prototype_cloning_traitorous/proc/AttachCloner(obj/machinery/clonepod/traitorous/pod)
+	if(!pod.connected)
+		pod.connected = src
+		LAZYADD(pods, pod)
+
+/obj/machinery/computer/prototype_cloning_traitorous/proc/DetachCloner(obj/machinery/clonepod/traitorous/pod)
+	pod.connected = null
+	LAZYREMOVE(pods, pod)
+
+/obj/machinery/computer/prototype_cloning_traitorous/attackby(obj/item/W, mob/user, params)
+	if(W.tool_behaviour == TOOL_MULTITOOL)
+		if(!multitool_check_buffer(user, W))
+			return
+		var/obj/item/multitool/P = W
+
+		if(istype(P.buffer, /obj/machinery/clonepod/traitorous))
+			if(get_area(P.buffer) != get_area(src))
+				to_chat(user, "<font color = #666633>-% Cannot link machines across power zones. Buffer cleared %-</font color>")
+				P.buffer = null
+				return
+			to_chat(user, "<font color = #666633>-% Successfully linked [P.buffer] with [src] %-</font color>")
+			var/obj/machinery/clonepod/traitorous/pod = P.buffer
+			if(pod.connected)
+				pod.connected.DetachCloner(pod)
+			AttachCloner(pod)
+		else
+			P.buffer = src
+			to_chat(user, "<font color = #666633>-% Successfully stored [REF(P.buffer)] [P.buffer.name] in buffer %-</font color>")
+		return
+	else
+		return ..()
+
+/obj/machinery/computer/prototype_cloning_traitorous/attack_hand(mob/user)
+	if(..())
+		return
+	interact(user)
+
+/obj/machinery/computer/prototype_cloning_traitorous/interact(mob/user)
+	user.set_machine(src)
+	add_fingerprint(user)
+
+	if(..())
+		return
+
+	updatemodules(TRUE)
+
+	var/dat = ""
+	dat += "<a href='byond://?src=[REF(src)];refresh=1'>Refresh</a>"
+
+	dat += "<h3>Cloning Pod Status</h3>"
+	dat += "<div class='statusDisplay'>[temp]&nbsp;</div>"
+
+	if (isnull(src.scanner) || !LAZYLEN(pods))
+		dat += "<h3>Modules</h3>"
+		//dat += "<a href='byond://?src=[REF(src)];relmodules=1'>Reload Modules</a>"
+		if (isnull(src.scanner))
+			dat += "<font class='bad'>ERROR: No Scanner detected!</font><br>"
+		if (!LAZYLEN(pods))
+			dat += "<font class='bad'>ERROR: No Pod detected</font><br>"
+
+	// Scan-n-Clone
+	if (!isnull(src.scanner))
+		var/mob/living/scanner_occupant = get_mob_or_brainmob(scanner.occupant)
+
+		dat += "<h3>Cloning</h3>"
+
+		dat += "<div class='statusDisplay'>"
+		if(!scanner_occupant)
+			dat += "Scanner Unoccupied"
+		else if(loading)
+			dat += "[scanner_occupant] => Scanning..."
+		else
+			scantemp = "Ready to Clone"
+			dat += "[scanner_occupant] => [scantemp]"
+		dat += "</div>"
+
+		if(scanner_occupant)
+			dat += "<a href='byond://?src=[REF(src)];clone=1'>Clone</a>"
+			dat += "<br><a href='byond://?src=[REF(src)];lock=1'>[src.scanner.locked ? "Unlock Scanner" : "Lock Scanner"]</a>"
+		else
+			dat += "<span class='linkOff'>Clone</span>"
+
+	var/datum/browser/popup = new(user, "cloning", "Prototype Cloning System Control")
```

Comment:
TGUI would be much better here

Reply:
Thanks, up to you

Reply:
As far as I know, all new UIs must be in TGUI3 to be accepted but crossed or someone will have to confirm that